src/storage/FileResourceStore.ts

```diff
@@ -0,0 +1,458 @@
+    }
+
+    const resultingAction = this.interactionHandler.getResultingAction(path, slug, linkType);
+    if (resultingAction.requestAction === RequestAction.CREATE_RESOURCE) {
+      // Create a file for the resource with as filepath the parent container.
+      return this.createFile(`${this.root}${resultingAction.parentContainer}`,
+        resultingAction.newIdentifier,
+        representation.data,
+        path.endsWith('/'),
+        metadata);
+    }
+
+    // Create a new container as subdirectory of the parent container.
+    return this.createContainer(`${this.root}${resultingAction.parentContainer}`,
+      resultingAction.newIdentifier,
+      path.endsWith('/'),
+      metadata);
+  }
+
+  public async deleteResource(identifier: ResourceIdentifier): Promise<void> {
+    let path = this.parseIdentifier(identifier);
+    if (path === '' || ensureTrailingSlash(path) === '/') {
+      throw new MethodNotAllowedHttpError('Cannot delete root container.');
+    }
+    path = `${this.root}${path}`;
+    return new Promise(async(resolve, reject): Promise<any> => {
+      try {
+        const stats = await fsPromises.lstat(path);
+        if (stats.isFile()) {
+          await fsPromises.unlink(path);
+          try {
+            await fsPromises.unlink(`${path}.metadata`);
+          } catch (_) {
+            // It's ok if there was no metadata file.
+          }
+          resolve();
+        } else if (stats.isDirectory()) {
+          path = ensureTrailingSlash(path);
+          const files = await fsPromises.readdir(path);
+          let match = files.find((file): any => !file.startsWith('.'));
+          if (match !== undefined) {
+            reject(new ConflictHttpError('Container is not empty.'));
+          }
+
+          match = files.find((file): any => file.startsWith('.'));
+          while (match) {
+            await fsPromises.unlink(`${path}${match}`);
+            const matchedFile = match;
+            files.filter((file): any => file !== matchedFile);
+            match = files.find((file): any => file.startsWith('.'));
+          }
+
+          await fsPromises.rmdir(path);
+          resolve();
+        } else {
+          reject(new NotFoundHttpError());
+        }
+      } catch (error) {
+        reject(new NotFoundHttpError());
+      }
+    });
+  }
+
+  /**
+   * Returns the stored representation for the given identifier.
+   * No preferences are supported.
+   * @param identifier - Identifier to retrieve.
+   *
+   * @returns The corresponding Representation.
+   */
+  public async getRepresentation(identifier: ResourceIdentifier): Promise<Representation> {
+    let path = `${this.root}${this.parseIdentifier(identifier)}`;
+    return new Promise(async(resolve, reject): Promise<any> => {
+      try {
+        const stats = await fsPromises.lstat(path);
+        if (stats.isFile()) {
+          const readStream = createReadStream(path);
+          const _contentType = contentType(path);
```

Comment:
Does the linter catch this?

Reply:
```suggestion
          const _contentType = getContentType(path);
```
?

Reply:
Was actually surprised the linter is ok with it, should maybe add id. The suggested change is not really clean though since `contentType` is a function imported from an external library.

Reply:
External libs do whatever they want, but we are free to rename the import 🙂

Reply:
The linter does not catch this.

`contentType()` is a function of mime-types, do you want me to import it as `getContentType()`?

Reply:
Yeah, or even `getContentTypeFromExtension`.

Reply:
Alright, will do.